File: samples/migrator/server/Migrator.Models/Entities/Product.cs
namespace Migrator.Models.Entities;
public class Product : Entity
{
    public int LegacyProductId { get; set; }
    public int LegacyModelId { get; set; }
    public int LegacySubCategoryId { get; set; }
    public int LegacyCategoryId { get; set; }
    public string Category { get; set; }
    public string Model { get; set; }
    public string Name { get; set; }
    public string ProductNumber { get; set; }
    public string SubCategory { get; set; }
}

File: samples/migrator/app/src/app/models/api/product.ts
import {
    FormBuilder,
    FormGroup,
    Validators
} from '@angular/forms';

import { Entity } from './base';

export interface Product extends Entity {
    legacyProductId: number;
    legacyModelId: number;
    legacySubCategoryId: number;
    legacyCategoryId: number;
    category: string;
    model: string;
    name: string;
    productNumber: string;
    subCategory: string;
}

export const GenerateProductForm = (product: Product, fb: FormBuilder): FormGroup =>
    fb.group({
        id: [product?.id ?? 0],
        legacyProductId: [product?.legacyProductId ?? 0],
        legacyModelId: [product?.legacyModelId ?? 0],
        legacySubCategoryId: [product?.legacySubCategoryId ?? 0],
        legacyCategoryId: [product?.legacyCategoryId ?? 0],
        category: [product?.category ?? '', Validators.required],
        model: [product?.model ?? '', Validators.required],
        name: [product?.name ?? '', Validators.required],
        productNumber: [product?.productNumber ?? '', Validators.required],
        subCategory: [product?.subCategory ?? '', Validators.required]
    });
